File: 2022/20221205/BinaryNumber/src/main/java/BinaryNumber.java
public class BinaryNumber {
    public int[] solution(String s) {
        int[] answer = new int[2];

        while (!s.equals("1")) {
            String removed = removeZero(s);
            answer[1] += s.length() - removed.length();

            s = toBinaryNumber(removed.length());
            answer[0] += 1;
        }

        return answer;
    }

    public String toBinaryNumber(int length) {
        return Integer.toBinaryString(length);
    }

    public String removeZero(String s) {
        return s.replaceAll("0", "");
    }
}

File: 2022/20221206/BinaryNumber.js
export default function solution(s) {
  const answer = Array(2).fill(0);

  while (s !== '1') {
    const removed = s.replace(/0/g, '');

    answer[1] += s.length - removed.length;

    s = removed.length.toString(2);
    answer[0] += 1;
  }
  return answer;
}
